### ISSUE TITLE
Enforce python naming convention 

### ISSUE THREAD

**Issue by eirikurj** [mdolab/.github#66]
# Description of feature
Our python code is mostly `CamelCase`, but there are exceptions. To keep the python MDO Lab codebase and APIs consistent and predicable, we could and probably should integrate some tools into our testing to enforce this.

# Potential solution
There are several tools available, but one option that fits into our current tools is to use the `flake8` extension, [pep8-naming](https://github.com/PyCQA/pep8-naming). 

Feel free to suggest other tools or approaches.

**Comment by eirikurj** [comment]
Yeah you are right, we are not really following PEP8 that well. They do however state in the [link](https://peps.python.org/pep-0008/#function-and-variable-names) you posted that
> mixedCase is allowed only in contexts where that’s already the prevailing style (e.g. threading.py), to retain backwards compatibility.

The tool does seem to support mixedCase and can maybe be of use just to get something going that is automated. I have however not tested the tool myself, but my main motivation was just to get some discussion started and seek out tools.
As you said , converting to `snake_case` is an option but would be a major change. Not sure if we want to do that unless there is a compelling reson and benefits of doing so.

**Comment by ewu63** [comment]
Yeah, I also think renaming all the variables will be fairly disruptive. However, I have not seen any linters which support non-PEP8 naming standards. For example, the one you linked (pep8-naming) will throw a ton of `N806` errors.
